frontend/src/utils/generate-warnings.ts

```diff
@@ -676,13 +689,64 @@ function produceSpecificCourseWarnings(
   warnings = warnings.concat(
     checkPrerequisites(term.classes, tracker, term.termId)
   );
+  ///// ADD CODE HERE TO CHECK IF THERE ARE DUPLICATES
+  warnings = warnings.concat(
+    checkDuplicates(term.classes, tracker, term.termId)
+  );
+  ////////////// END ///////////
+  return warnings;
+}
+
+/**
+ * Checks that the courses are not duplicates.
+ * @param toCheck The schedule to check.
+ * @param tracker tracker for courses taken
+ * @param termId the id for this term for the warning
+ */
+function checkDuplicates(
+  toCheck: ScheduleCourse[],
+  tracker: CourseTakenTracker,
+  termId: number
+): CourseWarning[] {
+  const warnings: CourseWarning[] = [];
+  for (let i = 0; i < toCheck.length; i++) {
+    const course = toCheck[i];
+    if (
+      courseCode(course) != "XXXX9999" &&
```

Comment:
actually, now that i think of it you should probably define an array of strings at the top of this file, that contains the names of courses that are placeholders